Radius.

HelenMamontova opened the pull request:
Function findUser added in the class Server of Radius plugin.

HelenMamontova committed:
Radius. Header file <mutex>, <jthread.hpp> added. Functions start, stop
declaration changed. Class members m_thread, m_mutex added. Function Run
declaration added.

HelenMamontova committed:
Radius. Header files added. Functions Start, Stop, Run definition added.

HelenMamontova committed:
Radius. New files server.h, server.cpp added.

HelenMamontova committed:
Radius. Definition of functions startReceive, makeResponse, handleSend,
handleReceive added.

HelenMamontova committed:
The Run function name fixed. Definition of secret variable sdded.

HelenMamontova committed:
The file server.cpp added to add_library command, command
find_package(OpenSSL 1.0.0 Required) added, OpenSSL::Crypto added to
target_link_libraries command in the block if(BUILD_MOD_RADIUS).

HelenMamontova committed:
Parameter token added to function Run() declaration.

HelenMamontova committed:
Radius. Method Start: variable isRunning=true removed, m_thread
definition changed. Method Stop: variable isRunning=false removed,
m_thread.joinable check added, isRunning check added, request_stop call added.
Method Run: parameter token added, variables secret, port removed,
object lock and isRunning=true added before cycle while,
cycle while added, isRunning=false added after cycle while.

HelenMamontova committed:
Radius. Hold the mutex removed,extra conditions
for m_thread.join() removed in the Stop function.

HelenMamontova committed:
Radius. Cycle while removed in function Run.

HelenMamontova committed:
Radius. The variables isRunning, errorStr replaced by m_running,
m_errorStr in class Radius.

HelenMamontova committed:
Radius. The variable isRunning replaced by m_running in function Run.

HelenMamontova committed:
Radius. Object name io_service changed to ioService in the function Run.

HelenMamontova committed:
Method SetRunning declaration added to the class RADIUS.

HelenMamontova committed:
Method SetRunning definition added. Method SetRunning call added to
function Run.

HelenMamontova committed:
Radius. Namespace STG added.

HelenMamontova committed:
Radius. Declaration using STG::Server added.

HelenMamontova committed:
Radius. Header file "stg/logger.h" added. RADIUS class member m_loger added.

HelenMamontova committed:
Radius. Header file "stg/common.h" added. Initialization of m_logger
added to constructor RADIUS. Output cerr replaced by logger and
printfd().

HelenMamontova committed:
Radius. Variable except added, messages fixed in the function Run.

HelenMamontova committed:
Radius. Thread join logic fixed in the function Stop.

HelenMamontova committed:
Keyword const added to std::lock_guard in the function SetRunning.

HelenMamontova committed:
Radius. Class member m_mutex moved to the top of list.

HelenMamontova committed:
Radius. Class member m_logger put after the m_running.

HelenMamontova committed:
Radius. Unnecessary variable except removed in the function Run.

HelenMamontova committed:
Radius. Extra whitespace removed in the function Stop. Function
IsRunning definition added.

HelenMamontova committed:
Radius. Function IsRunning declaration changed.

HelenMamontova committed:
Radius. Formatting fixed.

HelenMamontova committed:
Radius. Extra symbols '//' removed.

HelenMamontova committed:
Radius. Class member m_running initialization added to constructor
RADIUS.

HelenMamontova committed:
Radius.Header file "stg/module_settings.h" added. Class members variable
m_settings and method SetSettings added to class RADIUS.

HelenMamontova committed:
Radius. The parameters Secret, Dictionaries added, parameter Port
changed, other parameters removed.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Radius. Header files added. Class RAD_SETTINGS added. Methods
SetSettings, copy constructor RADIUS, assignment operator added. Method
ParseSettings changed. Formatting fixed.

HelenMamontova committed:
Radius. Parameter default changed to optional and parameters value
commented out for the parameters dictionaries, port.

HelenMamontova committed:
Radius. Description of parameters secret, port, dictionaries changed.

HelenMamontova committed:
Radius. Copy constructor RADIUS and assignment operator declaration
changed and moved to public section.

HelenMamontova committed:
Radius. Class member m_mutex moved to the top of section private.

HelenMamontova committed:
Radius. The m_port variable value initialization changed to 1812,
m_dictionaries variable initialization added to constructor RAD_SETTINGS.

HelenMamontova committed:
Radius. Check for missing parameters port and dictionaries removed in function ParseSettings.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Radius. The dist directory added.

HelenMamontova committed:
Radius. The methods declaration stop and start added.

HelenMamontova committed:
Radius. The methods stop, start definition added. Method start call
added to Server constructor.

HelenMamontova committed:
Radius. The header files "server.h", <boost/asio.hpp>, <memory> added.
The class members m_ioService, m_server added.

HelenMamontova committed:
Radius. The header files "server", <boost/asio.hpp>, <string> removed.
The method m_server->stop() call added to method Stop.

HelenMamontova committed:
Radius. Header files <stop_token> and <jthread.hpp> added. Parameter
token added to constructor Server. Class member m_token added.

HelenMamontova committed:
Radius. Parameter token and m_token class member initialization added to Server
constructor definition. The check m_token.stop_requested() added to
method handleSend before startReceive method call.

HelenMamontova committed:
Radius. The m_server class member initialization removed.

HelenMamontova committed:
Radius. The m_server->stop() call adnd m_thread.request_stop() call
reodered, the m_server check added before m_server->stop call in the
method Stop. Parameter token added to constructor Server in the function
Run.

HelenMamontova committed:
Radius. Header file "stg/logger.h" added. Class member m_logger added to
class Server.

HelenMamontova committed:
Radius. Header file "stg/common.h" added, header file <iostream>
removed. Class member m_logger initialization added to constructor
Server. Object std::cout replaced by function printfd and m_logger.

HelenMamontova committed:
Radius. Class member m_logger initialization changed in the constructor
Server. Check token moved before checking the error code in the
handleSend method.

HelenMamontova committed:
Radius. Check token added to handleReceive method.

HelenMamontova committed:
Radius. Header file <stop_token> replaced by <stop_token.hpp>, header
file <jthread.hpp> removed.

HelenMamontova committed:
Radius. Parameter m_logger added to constructor Server call in the
method Run.

HelenMamontova committed:
Radius. Parameter logger added to constructor Server declaration. Class
member m_logger added to class Server.

HelenMamontova committed:
Radius. Parameter logger added to constructor Server definition. Class
member m_logger initialization changed in the constructor Server.

HelenMamontova committed:
Radius. Extra whitespace removed ib the method Run.

HelenMamontova committed:
Radius. The order of operations changed in the method handleSend.

HelenMamontova committed:
Radius. The order of operations changed in the method handleReceive.

HelenMamontova committed:
Radius. Operator return removed when checking ec, operator else removed
when checking packet in the handleReceive method.

HelenMamontova committed:
Radius. Parameter token replaced by std::move(token) in the constructor Server
in method Run.

HelenMamontova committed:
Radius. Class member m_token initialization value changed to
std::move(token) in the constructor Server.

HelenMamontova committed:
Radius. The c_str() function added to message when checking error in the
handleSend and handleReceive methods.

HelenMamontova committed:
Radius. Header file <cstring> added.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Radius. Header files "stg/users.h", "stg/user.h", "stg/user_property.h"
added. The using statements added. The findUser, SetUsers functions
declaration added. Class member m_users added.

HelenMamontova committed:
Radius. Header files "radproto/attribute_types.h", <cstdint> added.
The m_users class member initialization added to constructor Server. The
findUser function definition added. The findUser function call added to
makeResponse function.

HelenMamontova committed:
Radius. Getting the values of login and password fixed in the function
findUser. Extra functions printfd call added.

HelenMamontova committed:
Radius. Using statements added. Method SetUsers added, variable class
member m_users added to class RADIUS.

HelenMamontova committed:
Radius. The class member variable m_users initialization added to
constructor Radius. Parameter m_users added to constructor Server call
in the method Run.

HelenMamontova committed:
Radius. Parameter users added to constructir Server declaration.
Function SetUsers removed fron class Server.

HelenMamontova committed:
Radius. Parameter users added to constructor Server definition. Variable
class member m_users initialization added to constructor Server.

HelenMamontova committed:
Radius. Function c_str added to login and password in m_logger and
printfd function in the function findUser.

HelenMamontova committed:
Merge remote-tracking branch 'madf/master' into radius

HelenMamontova committed:
Radius. Extra functions printfd removed.

madf commented on projects/stargazer/plugins/other/radius/radius.h:
```diff
@@ -18,6 +18,12 @@ namespace STG
+    class Users;
+
+    using UserPtr = STG::User*;
+    using ConstUserPtr = const User*;
```
`UserPtr` and `ConstUserPtr` are not used.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -48,6 +51,13 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
+        return RadProto::Packet(RadProto::ACCESS_REJECT, request.id(), request.auth(), attributes, vendorSpecific);
+    }
+
     if (request.type() == RadProto::ACCESS_REQUEST)
```
You should check request type before looking for a user.

madf commented on projects/stargazer/plugins/other/radius/server.h:
```diff
@@ -11,10 +14,16 @@
+    class Users;
+
+    using UserPtr = STG::User*;
+    using ConstUserPtr = const User*;
```
`UserPtr` and `ConstUserPtr` are not used.

madf commented on projects/stargazer/plugins/other/radius/server.h:
```diff
@@ -3,6 +3,9 @@
 #include "radproto/dictionaries.h"
+#include "stg/users.h"
+#include "stg/user.h"
+#include "stg/user_property.h"
```
`user.h` and `user_property.h` are not used in this file.

madf commented on projects/stargazer/plugins/other/radius/server.h:
```diff
@@ -3,6 +3,9 @@
 #include "radproto/socket.h"
 #include "radproto/packet.h"
 #include "radproto/dictionaries.h"
+#include "stg/users.h"
```
`users.h` is not used in this file, you forward-declare `Users` below.

madf commented on projects/stargazer/plugins/other/radius/server.h:
```diff
@@ -11,10 +14,16 @@
         public:
-            Server(boost::asio::io_service& io_service, const std::string& secret, uint16_t port, const std::string& filePath, std::stop_token token, PluginLogger& logger);
+            Server(boost::asio::io_service& io_service, const std::string& secret, uint16_t port, const std::string& filePath, std::stop_token token, PluginLogger& logger, STG::Users* users);
+            int findUser(const RadProto::Packet& packet);
```
I don't think `findUser` should be public.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -48,6 +51,13 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
     std::vector<uint8_t> vendorValue {0, 0, 0, 3};
     vendorSpecific.push_back(RadProto::VendorSpecific(m_dictionaries.vendorCode("Dlink"), m_dictionaries.vendorAttributeCode("Dlink", "Dlink-User-Level"), vendorValue));
 
+    if (findUser(request))
```
`findUser` returns `int`, but you use it like `bool`. Consider returning `bool` from `findUser`.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -48,6 +51,13 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
+    {
+        m_logger("Error findUser\n");
+        printfd(__FILE__, "Error findUser\n");
+        return RadProto::Packet(RadProto::ACCESS_REJECT, request.id(), request.auth(), attributes, vendorSpecific);
```
Reject packet should not have all those attributes.

madf replied:
Same below.

HelenMamontova committed:
Radius. The using declaration for UserPtr, ConstUserPtr removed.

HelenMamontova committed:
Radius. Looking for user moved after check request type in the
makeResponse function.

HelenMamontova committed:
Radius. The using declaration for UserPtr, ConstUserPtr removed. Header
files "stg/user.h", "stg/users.h", "stg/user.property.h" removed.

HelenMamontova committed:
Radius. Header files "stg/user.h", "stg/users.h" added.

HelenMamontova committed:
Radius. Function findUser moved to private.

HelenMamontova committed:
Radius. Return type of function findUser changed to bool.

HelenMamontova committed:
Radius. Return type of function findUser definition and call changed to bool.

HelenMamontova committed:
Radius. Variables attributesEmpty, vendorSpecificEmpty added for packet
ACCESS_REJECT in the function makeResponse.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -1,15 +1,20 @@
 using boost::system::error_code;
 
-Server::Server(boost::asio::io_service& io_service, const std::string& secret, uint16_t port, const std::string& filePath, std::stop_token token, PluginLogger& logger)
+Server::Server(boost::asio::io_service& io_service, const std::string& secret, uint16_t port, const std::string& filePath, std::stop_token token, PluginLogger& logger, STG::Users* users)
```
`Server` is in the `STG` namespace, so just `Users`, not `STG::Users`.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -48,10 +53,21 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
+    std::vector<RadProto::VendorSpecific> vendorSpecificEmpty;
 
-    return RadProto::Packet(RadProto::ACCESS_REJECT, request.id(), request.auth(), attributes, vendorSpecific);
+    if (request.type() == RadProto::ACCESS_REQUEST)
```
In cases like this:
```c++
if (condition)
{
    // A lot of code
    return something;
}
return someOtherThing;
```
consider turning inside out, like this:
```c++
if (!condition)
    return someOtherThing;
    
// A lot of code
return something;
```

It looks cleaner, it is easier to understand and it can reduce cyclomatic complexity.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -48,10 +53,21 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
+    {
+        if (findUser(request))
+            return RadProto::Packet(RadProto::ACCESS_ACCEPT, request.id(), request.auth(), attributes, vendorSpecific);
+        else
```
You do not need `else` after `return`.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -48,10 +53,21 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
+        {
+            m_logger("Error findUser\n");
+            printfd(__FILE__, "Error findUser\n");
+            return RadProto::Packet(RadProto::ACCESS_REJECT, request.id(), request.auth(), attributesEmpty, vendorSpecificEmpty);
```
If you need an empty vector, you don't have to create a dedicated variable. You can construct it in-place:
```c++
return RadProto::Packet(RadProto::ACCESS_REJECT, request.id(), request.auth(), {}, {});
```

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -48,10 +53,21 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
+            return RadProto::Packet(RadProto::ACCESS_REJECT, request.id(), request.auth(), attributesEmpty, vendorSpecificEmpty);
+        }
+    }
+    return RadProto::Packet(RadProto::ACCESS_REJECT, request.id(), request.auth(), attributesEmpty, vendorSpecificEmpty);
```
Same as above.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -84,5 +100,41 @@ void Server::handleReceive(const error_code& error, const std::optional<RadProto
+
+bool Server::findUser(const RadProto::Packet& packet)
+{
+    STG::User* user;
```
`Server` is in the `STG` namespace, you don't need to prefix `User` with `STG::`.

madf replied:
Also, please move it close to the point of use (`FindByName` call, I believe) and initialize with `nullptr`.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -84,5 +100,41 @@ void Server::handleReceive(const error_code& error, const std::optional<RadProto
+        return false;
+    }
+
+    printfd(__FILE__, "User FOUND!\n");
```
You've already said that the user is found, no need to repeat it twice.

HelenMamontova committed:
Radius. Type STG::Users* replaced by Users* in parameter users in the
constructor Server definition.

HelenMamontova committed:
Radius. Condition and code order changed when packet is returned in
makeResponse function.

HelenMamontova committed:
Radius. Unnecessary variables attributeEmpty, vendorSpecificEmpty
replaced by {} when ACCESS_REJECT packet is returned in makeResponse
function.

HelenMamontova committed:
Radius. Prefix STG of user object removed, user is initialized by
nullptr and moved to the point before call FindByName function in the
findUser function.

HelenMamontova committed:
Radius. Extra printfd function call removed in the function findUser.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -118,6 +111,7 @@ bool Server::findUser(const RadProto::Packet& packet)
             password = attribute->toString();
     }
 
+    User* user(nullptr);
```
`User* user = nullptr;`

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -48,10 +53,15 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
         return RadProto::Packet(RadProto::ACCESS_ACCEPT, request.id(), request.auth(), attributes, vendorSpecific);
 
-    return RadProto::Packet(RadProto::ACCESS_REJECT, request.id(), request.auth(), attributes, vendorSpecific);
+    m_logger("Error findUser\n");
```
We do not need to log it here, because we do it in the `findUser`.

madf commented on projects/stargazer/plugins/other/radius/radius.h:
```diff
@@ -74,7 +77,7 @@ namespace STG
 
             std::jthread m_thread;
-
+            STG::Users* m_users;
```
You don't need `STG::`.

madf commented on projects/stargazer/plugins/other/radius/server.h:
```diff
@@ -11,20 +11,24 @@
     {
         public:
-            Server(boost::asio::io_service& io_service, const std::string& secret, uint16_t port, const std::string& filePath, std::stop_token token, PluginLogger& logger);
+            Server(boost::asio::io_service& io_service, const std::string& secret, uint16_t port, const std::string& filePath, std::stop_token token, PluginLogger& logger, STG::Users* users);
```
You do not need `STG::`, just `Users`.

madf commented on projects/stargazer/plugins/other/radius/server.h:
```diff
@@ -11,20 +11,24 @@
 
             RadProto::Socket m_radius;
             RadProto::Dictionaries m_dictionaries;
+            STG::Users* m_users;
```
You do not need `STG::`.

HelenMamontova committed:
Radius. The way of initializing the object user changed in the findUser
function. Error log removed in the makeResponse function.

HelenMamontova committed:
Radius. Prefix STG of class member m_users and of parameter u in
SetUsers function removed.

HelenMamontova committed:
Radius. Prefix STG of class member m_users and of parameter users
in the constructor Server declaration removed.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -84,5 +93,38 @@ void Server::handleReceive(const error_code& error, const std::optional<RadProto
+    User* user = nullptr;
+    if (m_users->FindByName(login, &user))
+    {
+        m_logger("User's connect failed: user '%s' not found.", login.c_str());
```
'User's connect failed' is not what happens here. Just "User '%s' not found" would be ok, I think.

madf commented on projects/stargazer/plugins/other/radius/server.cpp:
```diff
@@ -84,5 +93,38 @@ void Server::handleReceive(const error_code& error, const std::optional<RadProto
+
+    if (password != user->GetProperties().password.Get())
+    {
+        m_logger("Password user is incorrect. %s", password.c_str());
```
Not 'Password user', but 'User's password'.

madf replied:
Same below.

HelenMamontova committed:
Radius. The messages of m_logger and printfd changed in function
findUser.
